x/auth/ante/sigverify.go

```diff
@@ -316,6 +319,16 @@ func (svd SigVerificationDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simul
 		if !simulate && !ctx.IsReCheckTx() {
 			anyPk, _ := codectypes.NewAnyWithValue(pubKey)
 
+			// TODO HV2: do we need to add here (and modify accordingly) the following code from heimdall?
+			// in that case, we need to implement the RecoverPubkey method
+			/*
+				p, err := authTypes.RecoverPubkey(signBytes, sig.Bytes())
```

Comment:
```suggestion
				var pk secp256k1.PubKeySecp256k1
				p, err := authTypes.RecoverPubkey(signBytes, sig.Bytes())
```

Reply:
There are a few things to consider here.
1. The [place where this is invoked in heimdall v1](https://github.com/maticnetwork/heimdall/blob/master/auth/ante.go#L238) uses two arguments `signBytes` and `sig.Bytes()` that we would need to match with the current implementation in heimdall v2. I need to understand what are those.
2. The `RecoveryPubkey` function is not [the one from cosmos package](https://github.com/0xPolygon/cosmos-sdk/blob/507ec1de34858aed7872e60a096f043a975ab3c9/crypto/keys/secp256k1/internal/secp256k1/secp256.go#L105), but a [custom one]
(https://github.com/maticnetwork/heimdall/blob/master/auth/types/txbuilder.go#L316) using `ethCrypto` library. I wonder if they behave the same. Also, the cosmos one is an internal function (not exported).